### PR TITLE
Integrate GCP Serving deployer with deployer interface

### LINKED ISSUE #143: Implement deployer interface for GCP serving deployer

### PR DESCRIPTION
**Pull request by rpasricha**
Fixes #143

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/189)

### CODE REVIEW

File: fairing/deployers/gcp/gcpserving.py [hunk 1]
```diff
@@ -5,27 +5,31 @@
 from googleapiclient import discovery
 from googleapiclient import errors
 
-# TODO: Integrate with DeployerInterface
-class GCPServingDeployer:
+# TODO: Implement predict and delete methods.
```
Comment (r2d4) [inline fairing/deployers/gcp/gcpserving.py]: Should predict and delete be methods on the DeployerInterface? Training jobs also use this interface.
Maybe

TrainingInterface, PredictionInterface -> DeployerInterface

SeldonPrediction -> PredictionInterface -> DeployerInterface
GCPPrediction -> PredictionInterface -> DeployerInterface

Comment (rpasricha) [reply fairing/deployers/gcp/gcpserving.py]: I wasn't referring to having those methods be part of the interface in this comment, but that hierarchy seems reasonable to me.

**Comment by karthikv2k** [comment]
created #217 as a follow up work to integrate into high level api.
/lgtm
/approve

**Comment by rpasricha** [comment]
Rebased PR to pull changes from master

**Pull request merged**